File: euchre.py
def update_score(current_score, called_trump, alone, tricks):
    team_1_score = current_score[0]
    team_2_score = current_score[1]
    
    score_count_1 = 0
    score_count_2 = 0
    for x in range(len(tricks)):
        if tricks[x] == 1:
            score_count_1 += 1
        else:
            score_count_2 += 1
    
    if called_trump == 1:
        if score_count_1  <= 2:
            team_2_score += 2
        elif score_count_1  < 5 and score_count_1 > 2:
            team_1_score += 1
        elif score_count_1 == 5 and alone == True:
            team_1_score += 4
        elif score_count_1 == 5 and alone == False:
            team_1_score += 2
            
    if called_trump == 2:
        if score_count_2 <= 2:
            team_1_score += 2
        elif score_count_2 < 5 and score_count_2 > 2:
            team_2_score += 1
        elif score_count_2 == 5 and alone == True:
            team_2_score += 4
        elif score_count_2 == 5 and alone == False:
            team_2_score += 2
            
    return [team_1_score, team_2_score]

File: euchre.js
function updateScore(currentScore, calledTrump, alone, tricks) {
    let team1Score = currentScore[0];
    let team2Score = currentScore[1];
    
    let scoreCount1 = 0;
    let scoreCount2 = 0;
    
    for (let i = 0; i < tricks.length; i++){
      if (tricks[i] == 1){
        scoreCount1++;
      }else{
        scoreCount2++;
      }
    }
    
  if (calledTrump == 1){
    if (scoreCount1 <= 2){
      team2Score += 2;
    }else if (scoreCount1 < 5 && scoreCount1 > 2){
      team1Score += 1;
    }else if (scoreCount1 == 5 && alone == true){
      team1Score += 4;
    }else if (scoreCount1 == 5 && alone == false){
      team1Score += 2;
    }
    }
  if (calledTrump == 2){
    if (scoreCount2 <= 2){
      team1Score += 2;
    }else if (scoreCount2 < 5 && scoreCount2 > 2){
      team2Score += 1;
    }else if (scoreCount2 == 5 && alone == true){
      team2Score += 4;
    }else if (scoreCount2 == 5 && alone == false){
      team2Score += 2;
    }
    }
  return [team1Score, team2Score];
  }